### PR TITLE
AD-1285 only send specified comments to middleware

### PR DESCRIPTION
**Pull request by benagain**
Solves #27 (Minimise interactions between Zendesk and Service Now).

### CODE REVIEW

File: src/SFA.DAS.Zendesk.Monitor.UnitTests/Tests/AuditedComment.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        public Comment AsComment => this;
+
+        public Audit AsAudit { get; }
+
+        public Event AuditTagEvent => AsAudit.Events[0]; 
+        
+        public new long? Id
+        {
+            get => base.Id;
+            set => base.Id = AsAudit.Events[^1].Id = value;
```
Comment (DannyGB) [inline src/SFA.DAS.Zendesk.Monitor.UnitTests/Tests/AuditedComment.cs]: [^1] is a new one on me

File: src/SFA.DAS.Zendesk.Monitor.UnitTests/Tests/Sharing_tickets.cs [hunk 2]
```diff
@@ -81,38 +82,120 @@ public async Task Sends_ticket_to_correct_endpoint([Frozen] FakeZendeskApi zende
         }
 
         [Theory, AutoDataDomain]
-        public async Task Sends_ticket_to_middleware_with_comments([Frozen] FakeZendeskApi zendesk, [Frozen] Middleware.IApi middleware, Watcher sut, [Pending.Solved] Ticket ticket, Comment[] comments)
+        public async Task Sends_nothing_to_middleware_when_ther_are_no_tagged_comments(
+            [Frozen] FakeZendeskApi zendesk,
+            [Frozen] Middleware.IApi middleware,
+            [Frozen] AuditedComment comment,
```
Comment (DannyGB) [inline src/SFA.DAS.Zendesk.Monitor.UnitTests/Tests/Sharing_tickets.cs]: Not an Autofixture expert at all but do these need to be Frozen? I thought the Frozen attribute passed the same instance to all further parameters of the same type and there is only the one parameter of each type?

File: src/SFA.DAS.Zendesk.Monitor/Zendesk/EventValueJsonConverter.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+
+namespace SFA.DAS.Zendesk.Monitor.Zendesk
+{
+    /// <summary>
+    /// Zendesk is a bit loose with their use of the "value" field in the 
+    /// "event" structure.  Sometimes it is a string, sometimes an array,
+    /// and sometimes an object.
```
Comment (DannyGB) [inline src/SFA.DAS.Zendesk.Monitor/Zendesk/EventValueJsonConverter.cs]: That's nice of them

File: src/SFA.DAS.Zendesk.Monitor/Zendesk/SharingTickets.cs [hunk 4]
```diff
@@ -21,13 +21,44 @@ public async Task<Option<SharedTicket>> GetTicketForSharing(long id)
...
+
             return sharing;
         }
 
+        private async Task LoadComments(SharedTicket response)
+        {
+            var a = await api.GetTicketComments(response.Response.Ticket);
+            var c = await api.GetTicketAudits(response.Response.Ticket);
+            var b = FilterComments(a, c);
+            response.Response.Comments = b;
```
Comment (DannyGB) [inline src/SFA.DAS.Zendesk.Monitor/Zendesk/SharingTickets.cs]: a, c and b are not great identifiers. Especially when a is used for Comments and c is used Audits

File: src/SFA.DAS.Zendesk.Monitor.UnitTests/Tests/AuditedComment.cs (hunk 1 above)
Comment (benagain) [reply src/SFA.DAS.Zendesk.Monitor.UnitTests/Tests/AuditedComment.cs]: "Index 1 from the end of the sequence" 

https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#indices-and-ranges

File: src/SFA.DAS.Zendesk.Monitor/Zendesk/SharingTickets.cs (hunk 4 above)
Comment (benagain) [reply src/SFA.DAS.Zendesk.Monitor/Zendesk/SharingTickets.cs]: Spot the quickly written test!

Good point, I missed a tidying up opportunity there.  Will rework.

Comment (benagain) [reply src/SFA.DAS.Zendesk.Monitor/Zendesk/SharingTickets.cs]: Updatd with better variable names.

**Pull request merged**